os/board/rtl8721csm/src/component/common/bluetooth/realtek/sdk/example/ble_central/ble_tizenrt_client.c

```diff
@@ -194,32 +195,29 @@ trble_result_e rtw_ble_client_stop_scan(void)
 }
 
 trble_result_e rtw_ble_client_connect(trble_conn_info* conn_info, bool is_secured_connect)
-{ 
-    if (conn_info == NULL || g_conn_req_num >= ble_app_link_table_size)
+{
+    if(ble_client_connect_is_running)
     {
-        printf("\r\n[%s] invalid\r\n", __FUNCTION__);
+        printf("\r\n[%s] ble_client_connect is running\r\n", __FUNCTION__);
         return TRBLE_FAIL;
-    }
+    } else
+        ble_client_connect_is_running = 1;
 
-    int i = 0;
-    while(i < ble_app_link_table_size && ble_tizenrt_bond_req_table[i++].addr) ;
-    ble_tizenrt_bond_req_table[i - 1].addr = os_mem_alloc(0, GAP_BD_ADDR_LEN);
-    if(ble_tizenrt_bond_req_table[i - 1].addr == NULL)
+    if(conn_info == NULL || g_master_link_num >= ble_app_link_table_size)
     {
-        debug_print("\n[%s] Memory allocation failed", __FUNCTION__);
+        printf("\r\n[%s] invalid\r\n", __FUNCTION__);
+        ble_client_connect_is_running = 0;
         return TRBLE_FAIL;
     }
-    memcpy(ble_tizenrt_bond_req_table[i - 1].addr, conn_info->addr.mac, GAP_BD_ADDR_LEN);
-    ble_tizenrt_bond_req_table[i - 1].is_secured_connect = is_secured_connect;
-    g_conn_req_num++;
-    debug_print("\r\n[%s] ble_tizenrt_bond_req_table[%d]\r\n", __FUNCTION__, i - 1);
+
+    memcpy(ble_tizenrt_bond_req_info->addr, conn_info->addr.mac, GAP_BD_ADDR_LEN);
```

Comment:
Is it okay not to check `ble_tizenrt_bond_req_info` is NULL or not?

Reply:
It is already allocated when the ble client was initialized.